File: app/widgets/molecular/dna_samples/collection_combo_box.rb
class Molecular::DnaSamples::CollectionComboBox < Templates::ComboBox
  def initialize options
    @dna_sample ||= options[:dna_sample] || fail('dna sample not given')
    @model_object = @dna_sample
    @attribute_path = 'collection'
    @value_method = 'id'
    @text_method = 'label'
    @search_method = 'collector'
    super
  end

  def value_searchtext
    #(value && ((value.respond_to?(value_method) && value.send(value_method)) || value[search_method])) || ''
    @dna_sample.try('collection').try('collector') || ''
  end

  def value_fulltext
    #(value && ((value.respond_to?(value_method) && value.send(value_method)) || value[search_method])) || ''
    @dna_sample.try('collection').try('collector') || ''
  end
end

File: app/widgets/molecular/dna_samples/collection_combo_box.js
//= require <templates/combo_box>

JooseModule('Molecular.DnaSamples', function () {
  JooseClass('CollectionComboBox', {
    isa: Templates.ComboBox,
    has: {
      object:  { is: 'ro', required: true, nullable: false },
      method: { is: 'ro', init: 'collection' },
      collectionURI: { is: 'ro', init: Route.forPathname('project_collections_path')},//'/projects/'+params['project_id']+'/collections' },
      valueMethod: { is: 'ro', init: 'id' },
      textMethod: { is: 'ro', init: 'label' },
      searchMethod: { is: 'ro', init: 'collector' },
      options: { is: 'ro', init: function () { return {
            requestHeaders: { Accept: 'text/javascript' },
            parameters: {
              select: 'id,label,collector,collection_number',
              order: 'collector, collection_number',
              include: '',
              limit: ''
            }} }},
      tagName: { init: 'dna_sample[collection_id]' }
    }
  })
});
